termcolor/src/lib.rs

```diff
@@ -954,6 +954,58 @@ impl<W: io::Write> Ansi<W> {
                 }
             }
         }
+        macro_rules! write_var_ansi_code {
```

Comment:
Some eyes on this would be good if its preferred over `format!`. I've added some test cases here but could have missed something.